activesupport/lib/active_support/core_ext/module/attribute_accessors.rb

```diff
@@ -8,6 +8,8 @@ class Module
   # Defines a class attribute and creates a class and instance reader methods.
   # The underlying class variable is set to +nil+, if it is not previously
   # defined.
+  # Note: Both methods will be public even if this method is called from
+  # inside a private or protected block.
```

Comment:
First: there's no such thing as a "private or protected block".

More generally, however, this feels like too strong of a 'gotcha' call-out. This is normal and expected behaviour: the helper defines a class/module method, and those are not affected by the current visibility scope in the class body. If we feel the need to make it explicit, I think the wording needs to lean more toward "contextual reminder" and less "beware".